From 95c3ad0174d9bb082699aa66c834de8802523e2b Mon Sep 17 00:00:00 2001
From: Bo Jiao <Bo.Jiao@mediatek.com>
Date: Thu, 7 Mar 2024 11:03:08 +0800
Subject: [PATCH] wifi: mt76: mt7915: only set MT76_MCU_RESET for the main phy

MT76_MCU_RESET is shared by all Phys, therefore it is only set for the main Phy.

Signed-off-by: Bo Jiao <Bo.Jiao@mediatek.com>
---
 mt7915/mac.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

--- a/mt7915/mac.c
+++ b/mt7915/mac.c
@@ -1338,10 +1338,8 @@ mt7915_mac_restart(struct mt7915_dev *de
 	set_bit(MT76_RESET, &dev->mphy.state);
 	set_bit(MT76_MCU_RESET, &dev->mphy.state);
 	wake_up(&dev->mt76.mcu.wait);
-	if (ext_phy) {
+	if (ext_phy)
 		set_bit(MT76_RESET, &ext_phy->state);
-		set_bit(MT76_MCU_RESET, &ext_phy->state);
-	}
 
 	/* lock/unlock all queues to ensure that no tx is pending */
 	mt76_txq_schedule_all(&dev->mphy);
